src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -451,4 +452,60 @@
      */
     <T extends Inventory> T query(Object... args);
 
+    /**
+     * Creates a new {@link Inventory.Builder} to build an {@link Inventory}.
+     *
+     * @return The builder
+     */
+    static Inventory.Builder builder() {
+        return Sponge.getRegistry().createBuilder(Inventory.Builder.class);
+    }
+
+    /**
+     * A Builder for Inventories based on {@link InventoryArchetype}s
+     */
+    interface Builder extends ResettableBuilder<Inventory, Builder> {
+
+        /**
+         * Sets the base {@link InventoryArchetype} for the Inventory
+         *
+         * @param archetype The InventoryArchetype
+         * @return Fluent pattern
+         */
+        Builder of(InventoryArchetype archetype);
+
+        /**
+         * Sets an {@link InventoryProperty}
+         * TODO only properties declared in the archetype are allowed? IllegalArguement?
+         *
+         * @param name The name
+         * @param property The property
+         * @return Fluent pattern
+         */
+        Builder property(String name, InventoryProperty property);
+
+        /**
+         * Sets the InventoryArchetype and Properties according to the {@link Carrier}s Inventory
+         *
+         * @param carrier The Carrier
+         * @return Fluent pattern
+         */
+        Builder forCarrier(Carrier carrier);
+
+        /**
+         * Sets the InventoryArchetype and Properties for a default Inventory of given {@link Carrier}
+         *
+         * @param carrier The Carrier class
+         * @return Fluent pattern
+         */
+        Builder forCarrier(Class<? extends Carrier> carrier);
```

Comment:
Why? It's returning a builder *for a `Carrier`* it's not adding a `Carrier` to a builder. It has the same semantic equivalence as `of` in other circumstances. If anything it should just be `for` except that it can't be because `for` is a keyword.